### ISSUE TITLE
 No such host is known

### ISSUE THREAD

**Issue by VikasJaiswal** [Azure/iotedge#1269]
I have deployed tempsensor module to IoTEdge Runtime, but when I check the logs, it is giving below error:

```
iotedge logs tempsensor
SimulatedTemperatureSensor Main() started.
Initializing simulated temperature sensor to send 500 messages, at an interval of 5 seconds.
To change this, set the environment variable MessageCount to the number of messages that should be sent (set it to -1 to send unlimited messages).
Information: Trying to initialize module client using transport type [Amqp_Tcp_Only].


Unhandled Exception: System.AggregateException: One or more errors occurred. (Transient network error occurred, please retry.) ---> Microsoft.Azure.Devices.Client.Exceptions.IotHubCommunicationException: Transient network error occurred, please retry. ---> System.Net.Sockets.SocketException: No such host is known
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpConnector.InitializeTransport(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpConnector.OpenConnectionAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpConnectionHolder.EnsureConnection(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpConnectionHolder.AmqpSessionCreator(DeviceIdentity deviceIdentity, ILinkFactory linkFactory, AmqpSessionSettings amqpSessionSettings, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpUnit.OpenAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpTransportHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ProtocolRoutingDelegatingHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.<>c__DisplayClass22_0.<<ExecuteWithErrorHandlingAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation)
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass32_0.<<OpenAsyncInternal>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.EnsureOpenedAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.OpenAsync()
   at ModuleLib.ModuleUtil.InitializeModuleClientAsync(TransportType transportType, ILogger logger) in C:\agent\_work\4\s\edge-modules\ModuleLib\ModuleUtil.cs:line 81
   at ModuleLib.ModuleUtil.CreateModuleClientAsync(TransportType transportType, ITransientErrorDetectionStrategy transientErrorDetectionStrategy, RetryStrategy retryStrategy, ILogger logger) in C:\agent\_work\4\s\edge-modules\ModuleLib\ModuleUtil.cs:line 41
   at SimulatedTemperatureSensor.Program.MainAsync() in C:\agent\_work\4\s\edge-modules\SimulatedTemperatureSensor\src\Program.cs:line 68
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at SimulatedTemperatureSensor.Program.Main() in C:\agent\_work\4\s\edge-modules\SimulatedTemperatureSensor\src\Program.cs:line 37
```

IoTEdge Version: 
iotedge 1.0.7.1

**Comment by myagley** [comment]
Hello, can you please run `iotedge check` on the device and post the result?

**Comment by VikasJaiswal** [comment]
Hi Myagley, 
Here is the result of  iotedge check

`PS C:\Windows\system32> iotedge check
Configuration checks
--------------------
v config.yaml is well-formed
v config.yaml has well-formed connection string
v container engine is installed and functional
v config.yaml has correct hostname
v config.yaml has correct URIs for daemon mgmt endpoint
v latest security daemon
v host time is close to real time
v container time is close to host time
? DNS server
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
? production readiness: certificates
    Device is using self-signed, automatically generated certs.
    Please see https://aka.ms/iotedge-prod-checklist-certs for best practices.
v production readiness: certificates expiry
v production readiness: container engine
? production readiness: logs policy
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.

Connectivity checks
-------------------
v host can connect to and perform TLS handshake with IoT Hub AMQP port
v host can connect to and perform TLS handshake with IoT Hub HTTPS port
v host can connect to and perform TLS handshake with IoT Hub MQTT port
v container on the IoT Edge module network can connect to IoT Hub AMQP port
v container on the IoT Edge module network can connect to IoT Hub HTTPS port
v container on the IoT Edge module network can connect to IoT Hub MQTT port
v Edge Hub can bind to ports on host

One or more checks raised warnings. Re-run with --verbose for more details.`

**I have tried to setup dns as well in the demon.json file of iotedge-moby. Below is the json string I have used** :
`{
    "dns": ["1.1.1.1"]
}`
But the issue is not resolved.
